x/fungible/keeper/gas_price.go

```diff
@@ -25,7 +25,10 @@ func (k Keeper) SetGasPrice(ctx sdk.Context, chainid *big.Int, gasPrice *big.Int
 		return 0, sdkerrors.Wrapf(types.ErrABIGet, "SystemContractMetaData")
 	}
 	res, err := k.CallEVM(ctx, *abi, types.ModuleAddressEVM, oracle, BigIntZero, big.NewInt(50_000), true, false, "setGasPrice", chainid, gasPrice)
-	if err != nil || res.Failed() {
+	if err != nil {
+		return 0, sdkerrors.Wrapf(types.ErrABIGet, err.Error())
```

Comment:
```suggestion
		return 0, sdkerrors.Wrapf(types.ErrContractCall, err.Error())
```
We can keep `ErrContractCall` since we make abstraction of `CallEVM` inner logic